src/avram/polymorphic.cr

```diff
@@ -0,0 +1,144 @@
+# Helpers for defining polymorphic associations
+module Avram::Polymorphic
+  # Sets up helper methods for validating and loading polymorphic associatiions
+  #
+  # This will generate methods for validiating, preloading, and lazy loading
+  # polymorphic associations.
+  #
+  # Let's start with this example:
+  #
+  # > *Important note:* the `belongs_to` associations must be nilable since only
+  # > one polymorphic can be set at a time. Remember to also make the association
+  # > nilable in the migration, for example: `add_belongs_to photo : Photo?`
+  #
+  # ```
+  # class Comment < BaseModel
+  #   table do
+  #     belongs_to photo : Photo?
+  #     belongs_to video : Video?
+  #     polymorphic :commentable, associations: [:photo, :video]
+  #   end
+  # end
+  #
+  # class Photo < BaseModel
+  #   table do
+  #     has_many comments : Comment
+  #   end
+  # end
+  #
+  # class Photo < BaseModel
+  #   table do
+  #     has_many comments : Comment
+  #   end
+  # end
+  # ```
+  #
+  # `has_many` calls work just like normal and so do `belongs_to`.
```

Comment:
For readability, I'd put both association types together up front:

```suggestion
  # `has_many` (and `belongs_to`) work just like normal.
```

Feel free to take or leave the parenthetical. I overuse them 😄

Reply:
Looks good to me!